src/Lean/Elab/Tactic/ElabTerm.lean

```diff
@@ -124,21 +124,20 @@ where
     let newMVarIds ← getMVarsNoDelayed val
     /- ignore let-rec auxiliary variables, they are synthesized automatically later -/
     let newMVarIds ← newMVarIds.filterM fun mvarId => return !(← Term.isLetRecAuxMVar mvarId)
-    let newMVarIds ← if allowNaturalHoles then
-      pure newMVarIds.toList
-    else
+    /- if we pass this check while `allowNaturalHoles` is `false`, then all mvars in `newMVarIds`
```

Comment:
Could you rephrase this comment? What is "this check" referring to?

Reply:
How's this as a rewrite of the comment? (Too verbose?)
```
    /- The following `unless … do` block guards against unassigned natural mvarids created during 
    `k` in the case that `allowNaturalMVars := false`. If we pass this block without aborting, we 
    can be assured that `newMVarIds` does not contain unassigned natural mvars created during `k`. 
    Note that in all cases we must allow `newMVarIds` to contain unassigned natural mvars which 
    were created *before* `k`; this is the purpose of `mvarCounterSaved`, which lets us distinguish 
    mvars created before `k` from those created during and after. See issue #2434. -/
```

Reply:
Tentatively committed, as I might be away from my computer for a while soon. :)